Update to qestdb 2.0.3 API

antst opened the pull request:
This implements qestdb-2.0 API and lift qestdb dependancy to questdb==2.0.3, as only 2.0.3 version of questdb python module client can be build by pip from sources (if Rust compiler "cargo" is installed).
Earlier versions have bugs here and there and PIP can not build them.
Also, it is much easier  to convince questdb people to provide binaries (to avoid building altogether)  for recent client, than for old one.

antst commented:
@CM000n please, look at it. As everyone, who upgraded to 2024.12, has problem currently.

mgumz commented:
@antst 

```
Logger: homeassistant.setup
Source: setup.py:269
First occurred: 07:04:57 (1 occurrences)
Last logged: 07:04:57

Setup failed for custom integration 'qss': Requirements for qss not found: ['questdb==2.0.3'].
```

I git-cloned your branch to `custom_components` (or rather only the `qss` subfolder). HA:2024.12.2

CM000n commented on custom_components/qss/manifest.json:
```diff
@@ -6,6 +6,6 @@
   "iot_class": "local_push",
   "issue_tracker": "https://github.com/CM000n/QSS/issues",
-  "requirements": ["questdb>=1.2,<2.0", "tenacity>=8.0"],
+  "requirements": ["questdb==2.0.3", "tenacity>=8.0"],
```
shouldn't we at least support all patch and minor version jumps instead of pinning a specific version?

mgumz replied:
@CM000n no, why? when you pin the dep … you do a release on which you know that it works. someone at questdb releases a breaking version, which results in issues opened here.

CM000n replied:
If they take semantic versioning seriously, minor patch releases should never break the API. With a pin, I see a greater risk that users will then remain on a lower version than currently available and not notice and use security-relevant patches.
Unfortunately, I don't have the time or inclination to publish a Qss release for every questdb patch release.

antst replied:
> shouldn't we at least support all patch and minor version jumps instead of pinning a specific version?

only 2.0.3 is actually useable and buildable for real :)
we can make it ">=2.0.3", hopping they will not break it again, but this is the most we can do

antst replied:
> If they take semantic versioning seriously, minor patch releases should never break the API. With a pin, I see a greater risk that users will then remain on a lower version than currently available and not notice and use security-relevant patches. Unfortunately, I don't have the time or inclination to publish a Qss release for every questdb patch release.

while 2.0.x should be compatible, but they have build system broken due to bugs. Asking for ">=2.0" can result in potential generation of problems.

CM000n commented on custom_components/qss/const.py:
```diff
@@ -14,7 +14,7 @@
 
 RETRY_WAIT_SECONDS = 5
-RETRY_ATTEMPTS = 10
+RETRY_ATTEMPTS = 10000
```
I don't know if I like this approach, I'll have to think about it again. 
But I think this change should be done independently with the changes necessary for the questdb client update.

antst replied:
Yes I can roll it back. But nothing wrong in retrying forever with 5 sec intervals. Much better than retrying for 50 sec and die, if your QestDB server takes longer to reboot, for example:)

CM000n commented:
Also `pre-commit` hooks are failing at the moment 😉

mgumz commented:
@antst works.

antst commented:
> hello @antst. Thank you very much for this PR 😊 Can you please also adjust the `pyproject.toml` file and update the lock file

I updated deps and version in project file and mainifest and incorporated changes which should make liniter happy. But I have zero clue about lock file. Keep in mind, I am not python dev :)

saintman23 commented:
Lookingforward for a fix to use QuestDB with latrest HA...

antst commented:
@CM000n 
I accidentally resolved conversation about dependency. But summary is simple: one can not intentionally put as dependency something which is broken :) And patch-releases  prior 2.0.3 are broken :)
